cppcache/src/ClientMetadataService.cpp

```diff
@@ -143,7 +143,13 @@ void ClientMetadataService::getClientPRMetadata(const char* regionFullPath) {
                                               reply.getFpaSet());
       if (m_bucketWaitTimeout > std::chrono::milliseconds::zero() &&
           reply.getNumBuckets() > 0) {
-        WriteGuard guard(m_PRbucketStatusLock);
+        std::unique_lock<std::timed_mutex> guard(m_timedBucketStatusLock,
+                                                 CLOSE_SERVICE_MUTEX_TIMEOUT);
+        if (!guard.owns_lock()) {
+          LOGERROR("Buckets statuses container read mutex timeout %lld seconds",
```

Comment:
Sorry you were given incorrect advice here. `std::chrono::duration` should not be assumed to convert to a `uint64_t`. The internal representation of the duration value is system dependent. No explicit conversion is defined in the std C++ spec to any primitive integer type. No std C++ method exists for duration to string conversions, though a a few custom options exist for converting `std::chrono::duration` to a string. I prefer one that uses our utility class for converting to a string representation with unit. 
```c++
LOGERROR("Buckets statuses container read mutex timeout %s", 
    util::chrono::duration::to_string(CLOSE_SERVICE_MUTEX_TIMEOUT).c_str());
```
Or using the `std::string` overload of `LOGERROR`:
```c++
LOGERROR("Buckets statuses container read mutex timeout " + 
    util::chrono::duration::to_string(CLOSE_SERVICE_MUTEX_TIMEOUT));
```
You may need to include "util/chrono/duration.hpp".

The above will produce log message like this:
```
Buckets statuses container read mutex timeout 60s.
```